extensions/parquet/base/src/main/java/io/deephaven/parquet/base/materializers/BigIntegerMaterializer.java

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.parquet.base.materializers;
+
+import io.deephaven.parquet.base.BigIntegerParquetBytesCodec;
+import io.deephaven.parquet.base.PageMaterializer;
+import io.deephaven.parquet.base.PageMaterializerFactory;
+import io.deephaven.util.codec.ObjectCodec;
+import org.apache.parquet.column.values.ValuesReader;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+
+public class BigIntegerMaterializer implements PageMaterializer {
+
+    public static final PageMaterializerFactory Factory = new PageMaterializerFactory() {
+
+        @Override
+        public PageMaterializer makeMaterializerWithNulls(ValuesReader dataReader, Object nullValue, int numValues) {
+            return new BigIntegerMaterializer(dataReader, (BigInteger) nullValue, numValues);
+        }
+
+        @Override
+        public PageMaterializer makeMaterializerNonNull(ValuesReader dataReader, int numValues) {
+            return new BigIntegerMaterializer(dataReader, numValues);
+        }
+    };
+
+    private static final ObjectCodec<BigInteger> CODEC = new BigIntegerParquetBytesCodec();
```

Comment:
We observed that this is not correct. You need to take the width into account.
I wonder if we want to use codec caching.